node/node.go

```diff
@@ -0,0 +1,75 @@
+// Copyright (c) 2022 IoTeX Foundation
+// This source code is provided 'as is' and no warranties are given as to title or non-infringement, merchantability
+// or fitness for purpose and, to the extent permitted by law, all liability for your use of the code is disclaimed.
+// This source code is governed by Apache License 2.0 that can be found in the LICENSE file.
+
+package node
+
+import (
+	"context"
+
+	"github.com/iotexproject/iotex-core/blockchain"
+	"github.com/iotexproject/iotex-core/consensus"
+	"github.com/iotexproject/iotex-core/p2p"
+	"github.com/iotexproject/iotex-core/pkg/lifecycle"
+	"github.com/iotexproject/iotex-core/pkg/routine"
+	"github.com/iotexproject/iotex-core/pkg/version"
+	"github.com/iotexproject/iotex-proto/golang/iotextypes"
+)
+
+type Node struct {
+	Addr    string
+	Height  uint64
+	Version string
+}
+
+// NodeManager manage nodes on the blockchain
+type NodeManager interface {
+	lifecycle.StartStopper
+	UpdateNode(*Node)
+	GetNode(addr string) *Node
+}
+
+type delegateManager struct {
+	cfg         Config
+	nodeMap     map[string]*Node
+	consensus   consensus.Consensus
+	broadcaster lifecycle.StartStopper
+	p2pAgent    p2p.Agent
+	bc          blockchain.Blockchain
```

Comment:
define two interfaces in this package to remove dependencies on external packages